src/analyses/callstringAnalysis.ml

```diff
@@ -0,0 +1,87 @@
+open Analyses
+open GoblintCil
+open GobConfig
+
+(* Specifies the type of the callstack elements for the callstring analysis*)
+module type Callstack_Type =
+sig
+  include CilType.S
+  val stackTypeName: string
+  val pushElem: fundec -> ('d,'g,'c,'v) ctx -> t option (* returns an elements that should be pushed to the Callstack *)
+end
+
+(** Analysis with the k-callsting approach, which uses the last k callstack elements as context.
+    With the CT argument it is possible to specify the type of the callstack elements*)
+module Spec (CT:Callstack_Type) : MCPSpec = 
+struct
+  include Analyses.IdentitySpec
+
+  (* simulates a call stack of depth k*)
+  module CallStack = struct
+    include Printable.PQueue (CT)
```

Comment:
Probably one could have an even more efficient data structure here given the size is known. I was thinking along the lines of a ring buffer:

https://ocaml.org/p/containers/2.8.1/doc/CCRingBuffer/module-type-S/index.html

(This basically boils down to an array & a pointer, but this is likely not the most pressing thing here)